Issue 4 aniM calculations

kiepczi opened the pull request:
As agreed in #96, work done on `issue_4` needed to be split into smaller chunks. 

In this PR, I propose the following:

- Adding `intervaltree` to `requirements.txt`. While working on the legacy `pyANI`, we settled on using this package to merge overlapping regions that are still present in `.deltafilter` files.
- `method_anim.py` is a script used to calculate aniM values, which was tested and found to be congruent with the legacy `pyANI` in `tests/methods/test_anim.py`.

kiepczi commented on pyani_plus/methods/method_anim.py:
```diff
@@ -0,0 +1,271 @@
+    :param deltafilter: Path to the filter file
+    :param indir: Path to input directory (FASTA files)
+    """
+    genome_lengths = {
```
This might not be needed if we decide to use `md5` hashes in the file names of comparisons, and not filenames (eg. `<genome_1_md5>_vs_<genome_2_md5>.filter` rather `<genome_1_vs_genome_2>.filter`).

peterjc commented:
I would be inclined to use ``tests/test_anim.py`` rather than ``tests/methods/test_anim.py`` (and did this for ``tests/test_fastani.py``).

peterjc commented on pyani_plus/methods/method_anim.py:
```diff
@@ -0,0 +1,254 @@
+
+# NOTE This might not be needed as this information can now
+# be stored in one of the dataframeas in the database.
+def get_genome_length(filename: Path) -> int:
```
You shouldn't need this but it will need some rearranging to have access to the DB at the time the value is used.

kiepczi replied:
Do you think that should be addressed under this branch now, or perhaps under a new one?

peterjc replied:
Probably a new PR - I suspect it will need some refactoring.

peterjc commented on pyani_plus/methods/method_anim.py:
```diff
@@ -0,0 +1,254 @@
+    :param filename: Path to the input .delta file
+
+    Calculates the aligned lengths for reference and query and average nucleotide
+    identity, and returns the cumulative total for each as a tuple.
```
Also returns similarity errors.

kiepczi commented:
Thanks @peterjc for the code review. 
I have now moved the `test_anim.py` from `tests/methods` to `tests` (4481a93), and updated the `parse_delta` doctoring (d8c680f).

peterjc commented on tests/test_anim.py:
```diff
@@ -0,0 +1,219 @@
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
```
Another doubled license block - perhaps the pre-commit hook needs tweaking (I think there is a fuzzy match option)

peterjc reviewed (changes requested):
I think once the double-license is fixed, we can merge this 👍

kiepczi commented:
> I think once the double-license is fixed, we can merge this 👍

Great! Merging now :)
